frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -185,6 +212,16 @@ export default function InteractiveWaveformModal({
     startTime: null,
     endTime: null,
   });
+  //state for peaks
+  const [peakSlope, setPeakSlope] = useState(Array(24).fill(0));
```

Comment:
Why are all 4 of these values needed for each? Can't we just store the two y values?